### DISCUSSION TITLE
WebGLTileLayer - is interpolation supposed to happen across tiles or not?

### DISCUSSION THREAD

**Discussion by kingkong-cmd** [openlayers/openlayers#16258] [Q&A] [answered]
I am using a WebGLTileLayer with interpolation of colors based on values of band 1 in png. The png's are prerendered to EPSG:3857 and tiled. I noticed that there are these glitches between tiles, ref screenshot below, and I believe it may be due to interpolation not happening across tiles. I've tried to check if there is something fishy happening in my tile creation, but cannot seem to find anything. 

So, I just wanted to check if it is expected behavior for interpolation not to happen across tiles?

```
import WebGLTileLayer from 'ol/layer/WebGLTile.js';

let layer = new WebGLTileLayer({
        source: xyz_source,
        transition:0,
        opacity:1,
        style: {
	color: [
	"interpolate",
		["linear"],
			["*", ["band", 1], 100],
                        [0, color_1],
                        [5, color_2],
                        .....
              ]
      }
});
```
![Screenshot from 2024-10-05 22-18-45](https://github.com/user-attachments/assets/a3def6e0-ed85-475e-b5bb-64e8b429fc33)

**Comment by mike-000** [comment, marked as answer]
There is no interpolation between tiles.  If the tiles have a gutter so the edges overlap there will be interpolation into the hidden gutter regions and the lack of interpolation between tiles will not be visible.  For example https://stackblitz.com/edit/js-ihsy4o?file=package.json,index.html,index.js

For an example with an XYZ source get an API key from https://www.nextzen.org/ and try

```
url: 'https://tile.nextzen.org/tilezen/terrain/v1/256/terrarium/{z}/{x}/{y}.png?api_key=your-nextzen-api-key',
tileSize: 256,
gutter: 0,

```
and

```
url: 'https://tile.nextzen.org/tilezen/terrain/v1/260/terrarium/{z}/{x}/{y}.png?api_key=your-nextzen-api-key',
tileSize: 256,
gutter: 2,
```

If you do not have a source with gutter available you could use a custom loader to buffer from neighbouring tiles, see #14964.

**Reply by kingkong-cmd** [reply]
All right, thank you.

My tiles are 256. Just so I understand, are you saying I could try making tiles 256 + x and then put `gutter: x `in the xyz-source to smooth things out? Assuming these glitches are in fact due to lack of interpolation between tiles, which I'm starting to second guess since the glitches are more pronounced than I would expect from interpolation issues alone.

Another solution would be to just set maxzoom to 0 and load only a single png for the entire world, but not sure if this strategy requires significantly more compute for rendering compared to tiling.
